sdk/monitor/azure-monitor-opentelemetry/azure/monitor/opentelemetry/diagnostics/_diagnostic_logging.py

```diff
@@ -0,0 +1,81 @@
+# license information.
+# --------------------------------------------------------------------------
+
+import logging
+import threading
+from os import makedirs
+from os.path import exists, join
+
+from azure.monitor.opentelemetry._constants import (
+    _CUSTOMER_IKEY,
+    _EXTENSION_VERSION,
+    _IS_DIAGNOSTICS_ENABLED,
+    _env_var_or_default,
+    _get_log_path,
+)
+from azure.monitor.opentelemetry._version import VERSION
+
+_DIAGNOSTIC_LOGGER_FILE_NAME = "applicationinsights-extension.log"
+_SITE_NAME = _env_var_or_default("WEBSITE_SITE_NAME")
+_SUBSCRIPTION_ID_ENV_VAR = _env_var_or_default("WEBSITE_OWNER_NAME")
+_SUBSCRIPTION_ID = (
+    _SUBSCRIPTION_ID_ENV_VAR.split("+")[0]
+    if _SUBSCRIPTION_ID_ENV_VAR
+    else None
+)
+_logger = logging.getLogger(__name__)
+_DIAGNOSTIC_LOG_PATH = _get_log_path()
+
+
+class AzureDiagnosticLogging:
+    _initialized = False
+    _lock = threading.Lock()
+    _f_handler = None
+
+    @classmethod
+    def _initialize(cls):
+        with AzureDiagnosticLogging._lock:
+            if not AzureDiagnosticLogging._initialized:
+                if _IS_DIAGNOSTICS_ENABLED and _DIAGNOSTIC_LOG_PATH:
+                    log_format = (
+                        "{"
+                        + '"time":"%(asctime)s.%(msecs)03d", '
+                        + '"level":"%(levelname)s", '
+                        + '"logger":"%(name)s", '
+                        + '"message":"%(message)s", '
+                        + '"properties":{'
+                        + '"operation":"Startup", '
+                        + f'"sitename":"{_SITE_NAME}", '
+                        + f'"ikey":"{_CUSTOMER_IKEY}", '
+                        + f'"extensionVersion":"{_EXTENSION_VERSION}", '
+                        + f'"sdkVersion":"{VERSION}", '
+                        + f'"subscriptionId":"{_SUBSCRIPTION_ID}", '
+                        + '"language":"python"'
+                        + "}"
+                        + "}"
+                    )
+                    if not exists(_DIAGNOSTIC_LOG_PATH):
+                        makedirs(_DIAGNOSTIC_LOG_PATH)
+                    AzureDiagnosticLogging._f_handler = logging.FileHandler(
+                        join(
+                            _DIAGNOSTIC_LOG_PATH, _DIAGNOSTIC_LOGGER_FILE_NAME
+                        )
+                    )
+                    formatter = logging.Formatter(
+                        fmt=log_format, datefmt="%Y-%m-%dT%H:%M:%S"
+                    )
+                    AzureDiagnosticLogging._f_handler.setFormatter(formatter)
+                    AzureDiagnosticLogging._initialized = True
+                    _logger.info("Initialized Azure Diagnostic Logger.")
+
+    @classmethod
+    def enable(cls, logger: logging.Logger):
+        AzureDiagnosticLogging._initialize()
+        if AzureDiagnosticLogging._initialized:
+            logger.addHandler(AzureDiagnosticLogging._f_handler)
```

Comment:
mypy parses this as assuming `_f_handler` can have type None which `addHandler` does not accept. Can we do a check if `_f_handler` is truthy in place of (or with) the `_initialized` check?

Reply:
Done